uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/LeaveBalanceChangeHelpersTrait.php

```diff
@@ -16,32 +16,39 @@ private function getBalanceChangeTypeValue($type) {
     return $this->balanceChangeTypes[$type];
   }
 
-  public function createSourcelessBalanceChange($entitlementID, $amount, $type) {
-    LeaveBalanceChange::create([
+  public function createSourcelessBalanceChange($entitlementID, $amount, $type, $expiryDate = null) {
+    $params = [
       'type_id' => $type,
       'entitlement_id' => $entitlementID,
       'amount' => $amount
-    ]);
+    ];
+
+    if($expiryDate) {
```

Comment:
It's fine to no have validation here. This is just a test helper method and it's not supposed to be used anywhere else. Also, if the date isn't in the expected format, the create() method will fail (also making the test fail)